### ISSUE TITLE
Issues with creating a new station 

### ISSUE THREAD

**Issue by dckwong** [mcrute/pydora#57]
Hi,
I'm currently using the APIClient class, that is your high level Pandora API client to search for something, then choose the best result from the results. After choosing the best result, I'd get, say a `chosen_station`, which can then use the `create_station()` function. It all seems to work, though I get a status of `fail`, with the message `An unexpected error occurred`, with `code 0`. I am sure that the resulting `chosen_station` is either a `SongSearchResultItem` or an `ArtistSearchResultItem`.

Any ideas what could be going wrong? I don't have any issues with using the client to get my own playlists or anything, but I'm unable to create stations from the search results.

**Comment by mcrute** [comment]
Hi @dckwong, could you post the code you used in your testing? Also are you using a free or paid account? The code below is a minimal test-case that I think does what you're attempting to do and I'm not able to reproduce the failure you're seeing. Error 0 is unfortunately not terribly meaningful as Pandora will throw it under a variety of conditions sometimes indicating server error and sometimes indicating a bad request.

As far as the `create_station` API goes it should always be safe to pass a search `token` as the `search_token` parameter regardless of if it is an artist, song, or genre. But it is never valid to accidentally pass, say, an artist token in the `track_token` parameter.

```python
from pandora.clientbuilder import PydoraConfigFileBuilder

client = PydoraConfigFileBuilder().build()
res = client.search("De/Vision")
station = client.create_station(res.artists[0].token)

print("Created station {stationName} with id {stationToken}".format(**station))
print([s for s in client.get_station_list() if s.token == station["stationToken"]])

client.delete_station(station["stationToken"])
```

I did spot a bug that the `create_station` API call doesn't return a model but just a raw `dict` and I'll get a fix in place for that but that's not the cause of your issue here.

If you can post your test code I'd be happy to help you debug further though.

**Comment by mcrute** [comment]
Just released 1.12.0 which returns a `Station` model for `create_station`, here's the updated test code if you're using that version.

```python
from pandora.clientbuilder import PydoraConfigFileBuilder

client = PydoraConfigFileBuilder().build()
res = client.search("De/Vision")
station = client.create_station(res.artists[0].token)

print([s for s in client.get_station_list() if s.token == station.token])

client.delete_station(station.token)
```

**Comment by mcrute** [comment]
I'm going to close this issue for now. If you're still experiencing issues with the station API please feel free to re-open it.
